MetasysServicesExampleApp/Program.cs

```diff
@@ -343,6 +349,67 @@ static void Main(string[] args)
+                getAlarmsForObject = Console.ReadLine();
+                args = getAlarmsForObject.Split(' ');
+
+                alarmFilterForObject.StartTime = args[0];
+                alarmFilterForObject.EndTime = args[1];
+                alarmFilterForObject.PriorityRange = args[2];
+                alarmFilterForObject.Type = Convert.ToInt32(args[3]);
+                alarmFilterForObject.ExcludePending = Convert.ToBoolean(args[4]);
+                alarmFilterForObject.ExcludeAcknowledged = Convert.ToBoolean(args[5]);
+                alarmFilterForObject.ExcludeDiscarded = Convert.ToBoolean(args[6]);
+                alarmFilterForObject.Attribute = Convert.ToInt32(args[7]);
+                alarmFilterForObject.Category = Convert.ToInt32(args[8]);
+                alarmFilterForObject.Page = Convert.ToInt32(args[9]);
+                alarmFilterForObject.PageSize = Convert.ToInt32(args[10]);
+                alarmFilterForObject.Sort = args[12];
+            }
+			
+            var alarmItemsForObject = client.GetAlarmsForAnObject(objectId, alarmFilterForObject);
+
+            Console.WriteLine("Enter network device id to get alarm details: ");
+            string networkDeviceId = Console.ReadLine();
+
+            Console.WriteLine(string.Format("\n Alarm details found for this network device {0}", networkDeviceId));
+
+            string getAlarmsForNetworkDevice;
+            AlarmFilter alarmFilterModelForNetworkDevice = new AlarmFilter();
+
+            if (args.Length != 12)
+            {
+                Console.WriteLine("Please enter these parameters separated by space: Start Time, End Time, Priority range, Type, Exclude pending, Exclude acknowledged, Exclude discarded, Attribute, Category, Page, Page Size, Sort" +
+                    "\nRefer to the metasys-server/basic-services-dotnet README if you need help getting started.");
+                getAlarmsForNetworkDevice = Console.ReadLine();
+                args = getAlarmsForNetworkDevice.Split(' ');
+
+                alarmFilterForObject.StartTime = args[0];
+                alarmFilterForObject.EndTime = args[1];
+                alarmFilterForObject.PriorityRange = args[2];
+                alarmFilterForObject.Type = Convert.ToInt32(args[3]);
+                alarmFilterForObject.ExcludePending = Convert.ToBoolean(args[4]);
+                alarmFilterForObject.ExcludeAcknowledged = Convert.ToBoolean(args[5]);
+                alarmFilterForObject.ExcludeDiscarded = Convert.ToBoolean(args[6]);
+                alarmFilterForObject.Attribute = Convert.ToInt32(args[7]);
+                alarmFilterForObject.Category = Convert.ToInt32(args[8]);
+                alarmFilterForObject.Page = Convert.ToInt32(args[9]);
+                alarmFilterForObject.PageSize = Convert.ToInt32(args[10]);
+                alarmFilterForObject.Sort = args[12];
+            }
```

Comment:
Before conversion you have to check that the value at the index of the argument should not be null otherwise conversion will throw an error. Add validation for the required field.